fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java

```diff
@@ -554,4 +568,24 @@ public static int computeMinGEPower2(int num) {
         num |= (num >>> 16);
         return num < 0 ? 1 : num + 1;
     }
+
+    public static boolean canEliminateNull(Set<ColumnRefOperator> nullOutputColumnOps, ScalarOperator expression) {
+        Map<ColumnRefOperator, ScalarOperator> m = nullOutputColumnOps.stream()
+                .map(op -> new ColumnRefOperator(op.getId(), op.getType(), op.getName(), true))
+                .collect(Collectors.toMap(identity(), col -> ConstantOperator.createNull(col.getType())));
+
+        for (ScalarOperator e : Utils.extractConjuncts(expression)) {
+            ScalarOperator nullEval = new ReplaceColumnRefRewriter(m).rewrite(e);
+
+            ScalarOperatorRewriter scalarRewriter = new ScalarOperatorRewriter();
+            // Call the ScalarOperatorRewriter function to perform constant folding
+            nullEval = scalarRewriter.rewrite(nullEval, ScalarOperatorRewriter.DEFAULT_REWRITE_RULES);
+            if (nullEval.isConstantRef() && ((ConstantOperator) nullEval).isNull()) {
```

Comment:
Why this can be eliminated?

Reply:
null means false, which means the expression "xxx is null" can never be true, so it can eliminate null